library/alloc/src/string.rs

```diff
@@ -3094,6 +3101,92 @@ impl fmt::Write for String {
     }
 }
 
+/// Placeholder docs.
+#[unstable(feature = "into_chars", reason = "new API", issue="0")]
+pub struct IntoChars {
+    bytes: Vec<u8>,
+}
+
+#[unstable(feature = "into_chars", reason = "new API", issue="0")]
+impl fmt::Debug for IntoChars {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_tuple("IntoChars").field(&self.as_str()).finish()
+    }
+}
+
+#[unstable(feature = "into_chars", reason = "new API", issue="0")]
+impl IntoChars {
+    /// Placeholder docs.
+    pub fn as_str(&self) -> &str {
+        // SAFETY: `bytes` is a valid UTF-8 string.
+        unsafe { str::from_utf8_unchecked(self.bytes.as_slice()) }
+    }
+
+    fn iter(&self) -> CharIndices {
+        self.as_str().char_indices()
+    }
+}
+
+#[unstable(feature = "into_chars", reason = "new API", issue="0")]
+impl AsRef<str> for IntoChars {
+    fn as_ref(&self) -> &str {
+        self.as_str()
+    }
+}
+
+#[unstable(feature = "into_chars", reason = "new API", issue="0")]
+impl AsRef<[u8]> for IntoChars {
+    fn as_ref(&self) -> &[u8] {
+        self.bytes.as_slice()
+    }
+}
+
+#[unstable(feature = "into_chars", reason = "new API", issue="0")]
+impl Iterator for IntoChars {
+    type Item = char;
+
+    #[inline]
+    fn next(&mut self) -> Option<char> {
+        let mut iter = self.iter();
+        match iter.next() {
+            None => None,
+            Some((_, ch)) => {
+                let offset = iter.offset();
+                drop(self.bytes.drain(..offset));
+                Some(ch)
+            }
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.iter().size_hint()
+    }
+
+    #[inline]
+    fn last(mut self) -> Option<char> {
+        self.next_back()
+    }
+}
+
+#[unstable(feature = "into_chars", reason = "new API", issue="0")]
+impl DoubleEndedIterator for IntoChars {
+    #[inline]
+    fn next_back(&mut self) -> Option<char> {
+        let mut iter = self.iter();
+        match iter.next_back() {
+            None => None,
+            Some((idx, ch)) => {
+                self.bytes.truncate(idx);
```

Comment:
I don't use `vec::IntoIter` as the snippet suggested at https://github.com/rust-lang/libs-team/issues/268#issuecomment-2472357143 because for `next_back`, we'd be dropping elements from the back. It's possible to _loop_ `self.bytes_iter.next_back()`, but using a Vec seems a net win.

Reply:
just use [`vec::IntoIter::advance_back_by`](https://doc.rust-lang.org/std/iter/trait.DoubleEndedIterator.html#method.advance_back_by)

Reply:
Thank you!